pkg/segment/writer/segstore.go

```diff
@@ -57,6 +57,11 @@ import (
 const MaxAgileTreeNodeCount = 8_000_000
 const colWipsSizeLimit = 2000 // We shouldn't exceed this during normal usage.
 
+const MaxConcurrentAgileTrees = 3
+
+var currentAgileTreeCount int
+var atreeCounterLock sync.Mutex = sync.Mutex{}
```

Comment:
Should you use a semaphore here instead?

Reply:
My goal is to not have any limits here, hence this temp code. If I do end up having limits then will switch to semaphore